common/perf-test-core/src/main/java/com/azure/perf/test/core/HttpPerfTestBase.java

```diff
@@ -0,0 +1,243 @@
+    // follow the standard convention, it can be configured manually using these fields.
+    protected final HttpClient httpClient;
+    protected final Iterable<HttpPipelinePolicy> policies;
+
+    private static final AtomicInteger GLOBAL_PARALLEL_INDEX = new AtomicInteger();
+    protected final int parallelIndex;
+
+    /**
+     * Creates an instance of the Http Based Performance test.
+     * @param options the performance test options to use while running the test.
+     */
+    public HttpPerfTestBase(TOptions options) {
+        super(options);
+        this.options = options;
+        this.parallelIndex = GLOBAL_PARALLEL_INDEX.getAndIncrement();
+
+        final SslContext sslContext;
+
+        if (options.isInsecure()) {
+            try {
+                sslContext = SslContextBuilder.forClient()
+                    .trustManager(InsecureTrustManagerFactory.INSTANCE)
+                    .build();
+            } catch (SSLException e) {
+                throw new IllegalStateException(e);
+            }
+
+            reactor.netty.http.client.HttpClient nettyHttpClient = reactor.netty.http.client.HttpClient.create()
+                .secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+
+            httpClient = new NettyAsyncHttpClientBuilder(nettyHttpClient).build();
+        } else {
+            sslContext = null;
+            httpClient = null;
+        }
+
+        if (options.getTestProxies() != null && !options.getTestProxies().isEmpty()) {
+            if (options.isInsecure()) {
+                recordPlaybackHttpClient = reactor.netty.http.client.HttpClient.create()
+                    .secure(sslContextSpec -> sslContextSpec.sslContext(sslContext));
+            } else {
+                recordPlaybackHttpClient = reactor.netty.http.client.HttpClient.create();
+            }
+
+            testProxy = options.getTestProxies().get(parallelIndex % options.getTestProxies().size());
+            testProxyPolicy = new TestProxyPolicy(testProxy);
+            policies = Arrays.asList(testProxyPolicy);
+        } else {
+            recordPlaybackHttpClient = null;
+            testProxy = null;
+            testProxyPolicy = null;
+            policies = null;
+        }
+    }
+
+    /**
+     * Attempts to configure a ClientBuilder using reflection.  If a ClientBuilder does not follow the standard convention,
+     * it can be configured manually using the "httpClient" and "policies" fields.
+     * @param clientBuilder The client builder.
+     * @throws IllegalStateException If reflective access to get httpClient or addPolicy methods fail.
+     */
+    protected void configureClientBuilder(Object clientBuilder) {
+        if (httpClient != null || policies != null) {
+            Class<?> clientBuilderClass = clientBuilder.getClass();
+
+            try {
+                if (httpClient != null) {
+                    Method httpClientMethod = clientBuilderClass.getMethod("httpClient", HttpClient.class);
```

Comment:
This may not work with JDK 9+ as in our module-info.java we don't open client builders for reflective access.

Reply:
Good point.
This PR just refactors this logic, as it was merged previously.
I will open a ticket to track this and we can investigate and make changes to support for JDK9+ as needed.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/26685